### DISCUSSION TITLE
Server Sent Events - Alternative for WebSockets?

### DISCUSSION THREAD

**Discussion by davidpiesse** [livewire/livewire#2360] [3. Feature ideas]
Server Sent Events (SSE) is a server push means to send data to the client.
[Wiki Link](https://en.wikipedia.org/wiki/Server-sent_events)

It is uni-directional usually and uses HTTP/2 protocols rather than websockets making it suitable by default compared to the install of a websocket solution.

I really like solutions like this rather than using Echo for simple apps.
I can't get a definitive answer on max connections / but it seems to be limited the same way NGINX is (1024).

There is an existing package - but it is a super simple implementation.
[PHP-SSE](https://github.com/hhxsv5/php-sse)

It eventually brings me back to the three options we have for getting events sent from the server to the client.
- Polling
- Websockets (Echo)
- SSE

For most prod app Echo & Pusher is going to be the most stable.

**Comment by nuernbergerA** [comment]
I already tinkered around with SSE before v2 launch. But it wasn't that great, due too connection limits.

For me polling is super effective:
Open a connection
Do some work
Return payload
Close connection

If you use some components on a single page and multiple that by concurrent users and you see where this ends.

But that was just my experience.
